File: test-suite-python/src/test/python/micronaut/docs/operator/OnAddFilter.py
# tag::reconciler[]
from java.util.function import Predicate
from jakarta.inject import Singleton

from io.kubernetes.client.openapi.models import V1ConfigMap


@Singleton
class OnAddFilter(Predicate[V1ConfigMap]):

    def test(self, v1_config_map: V1ConfigMap) -> bool:
        annotations = v1_config_map.getMetadata().getAnnotations()
        if annotations is not None:
            return "io.micronaut.operator" in annotations
        return False
# end::reconciler[]

File: test-suite/src/test/java/io/micronaut/docs/operator/OnAddFilter.java
package io.micronaut.docs.operator;
//tag::reconciler[]
import io.kubernetes.client.openapi.models.V1ConfigMap;
import jakarta.inject.Singleton;

import java.util.function.Predicate;

@Singleton
public class OnAddFilter implements Predicate<V1ConfigMap> {

    @Override
    public boolean test(V1ConfigMap v1ConfigMap) {
        if (v1ConfigMap.getMetadata().getAnnotations() != null) {
            return v1ConfigMap.getMetadata().getAnnotations().containsKey("io.micronaut.operator");
        }
        return false;
    }
}
//end::reconciler[]
